yjit/src/codegen.rs

```diff
@@ -1951,7 +1951,7 @@ fn gen_get_ivar(
     //       Eventually, we can encode whether an object is T_OBJECT or not
     //       inside object shapes.
     // too-complex shapes can't use index access, so we use rb_ivar_get for them too.
-    if !receiver_t_object || uses_custom_allocator || comptime_receiver.shape_too_complex() {
+    if !receiver_t_object || uses_custom_allocator || comptime_receiver.shape_too_complex() || (ctx.get_chain_depth() as i32) >= max_chain_depth {
```

Comment:
I generated this at `== max_chain_depth - 1` instead of `== max_chain_depth` (or `>=`) because I thought the C call block should also be counted as part of the number of chains. But if we think about the limit as the number of `jit_chain_guard` calls, then doing this at `== max_chain_depth` (or `>=`) makes sense :+1: